MdeModulePkg/Core/Dxe/Misc/MemoryProtectionSupport.c

```diff
@@ -3558,6 +3558,29 @@ UninstallMemoryAttributeProtocol (
   }
 }
 
+/**
+  Maps memory below 1MB (legacy BIOS memory) as readable, writeable, and executable.
+**/
+STATIC
+VOID
+MapLegacyBiosMemoryRWX (
+  VOID
+  )
+{
+  EFI_STATUS  Status;
+
+  if (gCpu != NULL) {
+    Status = gCpu->SetMemoryAttributes (
+                     gCpu,
+                     0x0,
+                     0x100000,

```

Comment:
To my understanding we should effectively be able to limit this from `0x0` to `0xA0000`. Where `0xA0000` to `0x100000` are hardware mapped ranges. In most cases, fixed MTRRs are used to set memory types for the Legacy BIOS range (detectable via the `FE` flag in the `IA32_MTRR_DEF_TYPE` MSR) and would produce ranges like:

```
WB:0000000000000000-000000000009FFFF
UC:00000000000A0000-00000000000BFFFF
WP:00000000000C0000-00000000000FFFF
```

Where the area below `0xA0000` would be outside the uncached video memory and write-protected BIOS flash areas. Fixed MTRRs aside though, I wouldn't expect code to be written and executed from that upper range. I mentioned this since part of the memory protection effort is minimizing privilege and this seems like an opportunity to do so.

Reply:
The EFI stub does indeed limit its search for usable memory for the trampoline code to 0x0-0xa0000, updating this PR to only set that range to match the MTRR WB range.

Reply:
Updated. I also included the link to the OS wiki page outlining the memory layout of legacy bios.